### PR TITLE
Improvement/template changes

### PR DESCRIPTION
**Pull request by lucieleonard**
Goal: continue to improve the theme.
What's changed: CSS, layout, search engine + results.

### CODE REVIEW

File: sphinx_scality/static/css/current.css_t [hunk 1]
```diff
@@ -247,21 +298,148 @@ div.footer div.leftwrapper {
...
+}
+
+/*Search results*/
+.term {
+    color: {{ theme_termcolor }};
+    font-weight: bold;
+}
+
+/*Search box*/
+input[type=text] {
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Oops, we need more specificity on this selector. So, two changes to do:

1. In `sphinx_scality/searchbox.html`:

```diff
 {%- if builder != 'singlehtml' %}
 <div role="search">
-  <form id="rtd-search-form" class="wy-form" action="{{ pathto('search') }}" method="get">
+  <form id="search-form" class="wy-form" action="{{ pathto('search') }}" method="get">
     <input type="text" name="q" placeholder="{{ _('Search docs') }}" />
     <input type="hidden" name="check_keywords" value="yes" />
     <input type="hidden" name="area" value="default" />
   </form>
 </div>
 {%- endif %}
```

2. Given this slight adjustment (not critical, but `rtd-` prefix is for ReadTheDocs :wink:), you can use a more specific selector, in this file:

```suggestion
form#search-form > input[name=q] {
```

File: sphinx_scality/static/css/current.css_t [hunk 2]
```diff
@@ -122,16 +123,15 @@ div.sidebar.left:before {
 
 div.sidebar.right {
     grid-area: rsidebar;
-    margin: 0 0% 0 -60%;
 }
 
 div.sidebar.right:before {
     content: '';
     position: absolute;
+    overflow
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Missing something here, no?

File: sphinx_scality/static/css/current.css_t [hunk 3]
```diff
@@ -384,27 +384,30 @@ div.body h6 {
 /* Admonition boxes */
 
 div.important {
-    background-color: {{ theme_importantboxbackground}};
+    background-color: {{ theme_boxbackground }};
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]: Isn't there a common class name / CSS selector to get _all_ box types? So you could share those `background-color` and `border-radius` properties (even `border` with a `border-color` override in subsequent rule definitions).

File: sphinx_scality/static/css/current.css_t (hunk 2 above)
Comment (lucieleonard) [reply sphinx_scality/static/css/current.css_t]: oops

File: sphinx_scality/static/js/documentation_options.js_t [hunk 4]
```diff
@@ -0,0 +1,25 @@
+var DOCUMENTATION_OPTIONS = {
```
[the remark this answers is not in the document]
Comment (lucieleonard) [reply sphinx_scality/static/js/documentation_options.js_t]: This one was added as-is, committed, and then renamed with changes - I think that's why it appears as it's just been added

File: sphinx_scality/static/css/current.css_t (hunk 3 above)
Comment (gdemonet) [reply sphinx_scality/static/css/current.css_t]:
```css
div.admonition {
  background-color: {{ theme_boxbackground }};
  border-radius: 4px;
  border: 1px solid red;
}

div.important {
  border-color: {{ theme_importantboxborder }};
}
```

File: sphinx_scality/static/js/searchtools.js [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+          });
+        }
+      }
+      // search finished, update title and status message
+      else {
+        Search.stopPulse();
+        Search.title.text(_('Search Results'));
+        if (!resultCount)
+        Search.status.html(_(`Your search did not match any documents. Try browsing the \
+        <a href="${KBLINK}">Knowledge Base</a> or contact Scality Support.`));
```
Comment (gdemonet) [inline sphinx_scality/static/js/searchtools.js]:
```suggestion
        <a href="${DOCUMENTATION_OPTIONS.KBLINK}">Knowledge Base</a> or contact Scality Support.`));
```

File: sphinx_scality/static/css/current.css_t [hunk 6]
```diff
@@ -18,8 +18,12 @@ body {
         "lsidebar content rsidebar"
         "footer footer footer";
     background-color: {{ theme_bgcolor }};
+    font-weight: 350;
```
[the remark this answers is not in the document]
Comment (lucieleonard) [reply sphinx_scality/static/css/current.css_t]: Tested: yes we need it! (without that the font is bolder)

File: sphinx_scality/static/css/current.css_t [hunk 7]
```diff
@@ -260,8 +315,137 @@ div.footer div.rightwrapper {
...
+
+div.admonition.tip {
+    border-color: {{ theme_tipboxborder }};
+}
+
+div.admonition.note {
+    border-color: {{ theme_noteboxborder }};
+}
+
+div.warning {
```
Comment (gdemonet) [inline sphinx_scality/static/css/current.css_t]:
```suggestion
div.admonition.warning {
```

**Pull request merged**